.github/workflows/toolkit-build-push.yml

```diff
@@ -0,0 +1,132 @@
+name: Reusable workflow to build and push multi-arch toolkit images
+
+on:
+  workflow_call:
+    inputs:
+      variant:
+        required: true
+        type: string
+      build_args:
+        required: false
+        type: string
+        default: ""
+      force:
+        required: false
+        type: boolean
+        default: false
+    secrets:
+      token:
+        required: true
+
+env:
+  TOOLKITS_PATH: toolkits
+
+jobs:
+  check-version:
+    name: Check if the version has been updated
+    outputs:
+      has_changed: ${{ steps.check.outputs.has_changed }}
+      version: ${{ steps.versions.outputs.new_version }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+
+      - id: versions
+        run: |
+          FILE="$TOOLKITS_PATH/VERSION.${{ inputs.variant }}"
+          PREV_VERSION=$(git show HEAD^:"$FILE" 2>/dev/null || echo "0.0.0")
+          NEW_VERSION=$(git show HEAD:"$FILE")
+
+          echo "prev_version=${PREV_VERSION}" >> $GITHUB_OUTPUT
+          echo "new_version=${NEW_VERSION}" >> $GITHUB_OUTPUT
+
+      - uses: aica-technology/.github/.github/actions/compare-versions@v1.0.2
+        id: check
+        with:
+          previous_version: ${{ steps.versions.outputs.prev_version }}
+          new_version: ${{ steps.versions.outputs.new_version }}
+
+  metadata:
+    needs: check-version
+    if: ${{ needs.check-version.outputs.has_changed == 'true' || inputs.force }}
+    runs-on: ubuntu-latest
+    name: Metadata
+    outputs:
+      image_name: ${{ steps.ensure-image.outputs.image_name }}
+      image_tags: ${{ steps.tags.outputs.image_tags }}
+      build_flags: ${{ steps.tags.outputs.build_flags }}
+      git_tag: ${{ steps.tags.outputs.git_tag }}
+    steps:
+      - uses: aica-technology/.github/.github/actions/ghcr-ensure-prefix@v0.6.0
+        id: ensure-image
+        with:
+          image_name: aica-technology/${{ inputs.variant }}-toolkit # TODO: that's my original naming, but perhaps toolkit-${{ inputs.variant }} would be better?
```

Comment:
I agree with the comment, maybe even `aica-technology/toolkits/${{ inputs.variant }}` if that works?